core/internal/cltest/client.go

```diff
@@ -0,0 +1,328 @@
+package cltest
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/core/assets"
+	"github.com/smartcontractkit/chainlink/core/eth"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// SimulatedBackendClient is an eth.SimulatedBackendClient implementation using
+// a simulated blockchain backend. Note that not all RPC methods are implemented
+// here.
+type SimulatedBackendClient struct {
+	b       *backends.SimulatedBackend
+	t       testing.TB
+	chainId int
+}
+
+// Close terminates the underlying blockchain's update loop.
+func (c *SimulatedBackendClient) Close() {
+	c.b.Close()
+}
+
+var _ eth.Client = (*SimulatedBackendClient)(nil)
+
+// checkEthCallArgs extracts and verifies the arguments for an eth_call RPC
+func (c *SimulatedBackendClient) checkEthCallArgs(
+	args []interface{}) (*eth.CallArgs, *big.Int, error) {
+	if len(args) != 2 {
+		return nil, nil, fmt.Errorf(
+			"should have two arguments after \"eth_call\", got %d", len(args))
+	}
+	callArgs, ok := args[0].(eth.CallArgs)
+	if !ok {
+		return nil, nil, fmt.Errorf("third arg to SimulatedBackendClient.Call "+
+			"must be an eth.CallArgs, got %+#v", args[0])
+	}
+	blockNumber, err := c.blockNumber(args[1])
+	if err != nil || blockNumber.Cmp(c.currentBlockNumber()) != 0 {
+		return nil, nil, fmt.Errorf("fourth arg to SimulatedBackendClient.Call "+
+			"must the string \"latest\", or a *big.Int equal to current "+
```

Comment:
"must be the string"

Reply:
Oops, missed this before merging. Thanks.